lib/kitchen/provisioner/ansible_playbook.rb

```diff
@@ -401,25 +401,29 @@ def run_command
         cmd = "#{cd_ansible} #{cmd}" if !config[:ansible_sudo].nil? && !config[:ansible_sudo]
         cmd = "#{copy_private_key_cmd} #{cmd}" if config[:private_key]
 
-        result = [
-          cmd,
-          ansible_inventory_flag,
-          ansible_limit_flag,
-          ansible_connection_flag,
-          "-M #{File.join(config[:root_path], 'modules')}",
-          ansible_verbose_flag,
-          ansible_check_flag,
-          ansible_diff_flag,
-          ansible_vault_flag,
-          private_key,
-          extra_vars,
-          extra_vars_file,
-          tags,
-          ansible_extra_flags,
-          "#{File.join(config[:root_path], File.basename(config[:playbook]))}"
-        ].join(' ')
+        def _run(cmd, idempotence = false)
```

Comment:
I felt silly naming this method (and writing it), but I wanted to avoid duplicating it while still allowing the generation of `tags` to be dynamic.

Thoughts/criticisms welcome.

Reply:
Whats the impact of this change?

Reply:
When i get a moment i will review code. If its doesn't have any impact on the product no problem i will merge. i not then we need a separate parameter like idempotency_test_with_tags or something.

Reply:
There is no impact to the main `result` command, and no impact to the `result` command used for idempotence either _UNLESS_ one of the new flags is specified:

```
idempotencty_tags: ...
idempotencty_skip_tags: ...
```

I did this to share the command assembly code, but can split it out to prevent from modifying the existing code paths if you'd prefer!